manifests/blacklist.pp

```diff
@@ -15,10 +15,13 @@
   $ensure=present,
   $file='/etc/modprobe.d/blacklist.conf',
 ) {
-  ::kmod::generic {"blacklist ${name}":
-    ensure => $ensure,
-    type   => 'blacklist',
-    module => $name,
-    file   => $file,
+
+
+  kmod::setting { "kmod::blacklist ${title}":
+    ensure   => $ensure,
+    module   => $title,
```

Comment:
I'd use `$name` instead of title here, in case the end user wants to take advantage of the namevar.